widgets/gauge/gauge_test.go

```diff
@@ -841,6 +841,100 @@ func TestGauge(t *testing.T) {
 				return ft
 			},
 		},
+		{
+			desc: "threshold with border percentage",
+			opts: []Option{
+				Char('o'),
+				Threshold(20, linestyle.Double),
+			},
+			percent: &percentCall{p: 35},
+			canvas:  image.Rect(0, 0, 10, 3),
+			want: func(size image.Point) *faketerm.Terminal {
+				ft := faketerm.MustNew(size)
+				c := testcanvas.MustNew(ft.Area())
+
+				testdraw.MustRectangle(c, image.Rect(0, 0, 3, 3),
+					draw.RectChar('o'),
+					draw.RectCellOpts(cell.BgColor(cell.ColorGreen)),
+				)
+				testdraw.MustText(c, "35%", image.Point{3, 1})
+				testdraw.MustHVLines(c, []draw.HVLine{{
+					Start: image.Point{X: 2, Y: 0},
+					End:   image.Point{X: 2, Y: 2},
+				}}, draw.HVLineStyle(linestyle.Double))
+				testcanvas.MustApply(c, ft)
+				return ft
+			},
+		},
+		{
+			desc: "threshold without border absolute",
+			opts: []Option{
+				Char('o'),
+				Threshold(3, linestyle.Light, cell.BgColor(cell.ColorRed)),
+				Border(linestyle.None),
+				HideTextProgress(),
+			},
+			absolute: &absoluteCall{done: 4, total: 10},
+			canvas:   image.Rect(0, 0, 10, 3),
+			want: func(size image.Point) *faketerm.Terminal {
+				ft := faketerm.MustNew(size)
+				c := testcanvas.MustNew(ft.Area())
+
+				testdraw.MustRectangle(c, image.Rect(0, 0, 3, 3),
+					draw.RectChar('o'),
+					draw.RectCellOpts(cell.BgColor(cell.ColorGreen)),
+				)
+				testdraw.MustHVLines(c, []draw.HVLine{{
+					Start: image.Point{X: 3, Y: 0},
+					End:   image.Point{X: 3, Y: 2},
+				}}, draw.HVLineStyle(linestyle.Light),
+					draw.HVLineCellOpts(cell.BgColor(cell.ColorRed)))
+				testcanvas.MustApply(c, ft)
+				return ft
+			},
+		},
+		{
+			desc: "threshold outside of bounds (negative)",
+			opts: []Option{
+				Char('o'),
+				HideTextProgress(),
+				Threshold(-1, linestyle.Light), // ignored
+			},
+			percent: &percentCall{p: 35},
+			canvas:  image.Rect(0, 0, 10, 3),
+			want: func(size image.Point) *faketerm.Terminal {
+				ft := faketerm.MustNew(size)
+				c := testcanvas.MustNew(ft.Area())
+
+				testdraw.MustRectangle(c, image.Rect(0, 0, 3, 3),
+					draw.RectChar('o'),
+					draw.RectCellOpts(cell.BgColor(cell.ColorGreen)),
+				)
+				testcanvas.MustApply(c, ft)
+				return ft
+			},
+		},
+		{
+			desc: "threshold outside of bounds (>=max)",
```

Comment:
Can we also add a test cases based on where the progress is relative to the threshold. E.g.:

- progress below threshold
- progress exactly at threshold
- progress after threshold